src/eth/Cdp.js

```diff
@@ -35,13 +37,30 @@ export default class Cdp {
       .get('web3')
       .currentAccount();
 
-    const getId = new Promise(resolve => {
-      tubContract.onlognewcup = function(address, cdpIdBytes32) {
-        if (currentAccount.toLowerCase() == address.toLowerCase()) {
-          this.removeListener();
-          resolve(ethersUtils.bigNumberify(cdpIdBytes32).toNumber());
-        }
-      };
+    const getId = new Promise(async resolve => {
+      let log = {};
+      if (this._web3Service.usingWebsockets()) {
+        const sai_tub = this._smartContractService._getContractInfo(
```

Comment:
Will do. Just one question I had when implementing this. I wanted to access this `_getContractInfo` function from inside the event subscription function (`subscribeLog`/`waitForMatchingEvent`) so it only needed to call the contract name and event name from that contract.

```javascript
log = await this._web3Service.subscribeLog(contracts.SAI_TUB, 'LogNewCup');
```

This would further increase the complexity of the `subscribeLog` function but would make calling an event into a one-liner. This could be untenable as I was unable to figure out how to access the smartContractSservice from the web3Service

Reply:
Yeah, that would be cleaner. But the way the dependency injection system is now, it would create a circular dependency. Ethan was running into a similar issue yesterday which makes me think it's worth looking into changing the dependency system to allow this somehow.